app/services/puppet_fact_parser.rb

```diff
@@ -40,38 +40,38 @@ def operatingsystem
     end
   end
 
-  def environment
+  def environment_name
     # by default, puppet doesn't store an env name in the database
-    name = facts[:environment] || facts[:agent_specified_environment] || Setting[:default_puppet_environment]
-    Environment.unscoped.where(:name => name).first_or_create
+    facts[:environment] || facts[:agent_specified_environment] || Setting[:default_puppet_environment]
   end
 
-  def architecture
+  def architecture_name
     # On solaris and junos architecture fact is hardwareisa
     name = case os_name
              when /(sunos|solaris|junos)/i
-               facts[:hardwareisa]
+               hardware_isa
              else
-               facts[:architecture] || facts[:hardwareisa]
+               architecture_fact || hardware_isa
            end
-    # ensure that we convert debian legacy to standard
+    # Normalize some output, like on Debian and FreeBSD
     name = "x86_64" if name == "amd64"
     name = "aarch64" if name == "arm64"
-    Architecture.where(:name => name).first_or_create if name.present?
+    name
   end
 
-  def model
-    name = facts[:productname] || facts[:model] || facts[:boardproductname]
+  def model_name
+    # TODO: not sure where model comes from, not Facter
```

Comment:
could it be something in FDI? perhaps that was a custom fact, that should have been prefixed with `foreman_`, we may need to search in git history

Reply:
It was added in 8ac3ea75454ef2bc105c052e76383f3ceaf82232 way back in 2010 as part of https://projects.theforeman.org/issues/289. I don't think FDI existed back then. My guess is that it was either a custom fact or a bug because it meant the fact `hardwaremodel` (which did exist back in the released Facter version at the time, which was 1.5).

Reply:
This does not come from FDI. No idea, let's just remove these.